src/chain/stateTransition/index.ts

```diff
@@ -8,35 +8,43 @@ import {
 } from "../../types";
 
 import {processBlock} from "./block";
-import {processEpoch, shouldProcessEpoch} from "./epoch";
-import {advanceSlot, cacheState} from "./slot";
+import {processEpoch} from "./epoch";
+import {processSlots} from "./slot";
+import {hashTreeRoot} from "@chainsafe/ssz";
+import  assert from "assert";
+
 
 export {
-  advanceSlot,
-  cacheState,
   processBlock,
   processEpoch,
 };
 
-export function executeStateTransition(state: BeaconState, block: BeaconBlock | null, verifyStateRoot = true): BeaconState {
-
-  if (block) {
-    while(state.slot < block.slot) {
-      cacheState(state);
-      if (shouldProcessEpoch(state)) {
-        processEpoch(state);
-      }
-      advanceSlot(state);
-    }
-    processBlock(state, block, verifyStateRoot);
-  } else {
-    cacheState(state);
-    if (shouldProcessEpoch(state)) {
-      processEpoch(state);
-    }
-    advanceSlot(state);
-  }
+//SPEC 0.7
+// def state_transition(state: BeaconState, block: BeaconBlock, validate_state_root: bool=False) -> BeaconState:
+//   # Process slots (including those with no blocks) since block
+// process_slots(state, block.slot)
+// # Process block
+// process_block(state, block)
+// # Validate state root (`validate_state_root == True` in production)
+// if validate_state_root:
+// assert block.state_root == hash_tree_root(state)
+// # Return post-state
+// return state
 
 
+export function stateTransition(
+  state: BeaconState, block: BeaconBlock,
+  validateStateRoot = false
+): BeaconState {
+  // Process slots (including those with no blocks) since block
+  processSlots(state, block.slot);
+  // Process block
+  processBlock(state, block)
+  // Validate state root (`validate_state_root == True` in production)
+  if (validateStateRoot){
+    assert(block.stateRoot == hashTreeRoot(state, BeaconState));
```

Comment:
use equals method

Reply:
We'll need to update `@chainsafe/ssz` to v0.4.2

Reply:
Make an issue :)

Reply:
oih nvm thats already deployed